test/code/providers/runas_provider/runasprovider_test.cpp

```diff
@@ -101,6 +102,17 @@ class SCXRunAsProviderTest : public CPPUNIT_NS::TestFixture
             configurator->SetAllowRoot(true);
         }
         SCXCore::g_RunAsProvider.SetConfigurator(configurator);
+        // Create tmpdir link for execute script tests
+        int err = system("mkdir -p /etc/opt/microsoft/scx/conf > /dev/null 2>&1");
+        err = system("ln -s /tmp /etc/opt/microsoft/scx/conf/tmpdir > /dev/null 2>&1");
+        // Needed to add this as system() calls were giving unused variable warning causing 
+        // the compilation to fail
+        if(err) 
+        { 
+        	std::ostringstream os;
+        	os << "Unexpected error in setup: " << SCXCoreLib::strerror(err & 127);
+        	std::cout<<os.str()<<std::endl;
+        }
```

Comment:
This is yucky. I don't think you should be updating system locations, you may mess with a kit that I already have installed.

I suggest you use dependency injection to write somewhere in the intermediate directory. That way, the calling code is still tested, but you're not bound to a specific location. That would work nicely too, where you set the directory path on a per-project basis in my PAL recommendation. That way, dependency injection would just set to the test location in unit tests.